src/main/java/edu/cornell/kfs/sys/batch/MoveFileStep.java

```diff
@@ -0,0 +1,128 @@
+ */
+public class MoveFileStep extends AbstractStep {
+
+    private static final org.apache.log4j.Logger LOG = org.apache.log4j.Logger.getLogger(MoveFileStep.class);
+
+    protected String sourcePath;
+    protected String targetPath;
+    protected Pattern fileNamePattern;
+
+    @Override
+    public void afterPropertiesSet() throws Exception {
+        if (StringUtils.isBlank(sourcePath)) {
+            throw new IllegalStateException("sourcePath cannot be blank");
+        } else if (StringUtils.isBlank(targetPath)) {
+            throw new IllegalStateException("targetPath cannot be blank");
+        } else if (fileNamePattern == null) {
+            throw new IllegalStateException("fileNamePattern cannot be null");
+        }
+        
+        super.afterPropertiesSet();
+    }
+
+    @Override
+    public boolean execute(String jobName, Date jobRunDate) throws InterruptedException {
+        return moveFiles();
+    }
+
+    protected boolean moveFiles() {
+        verifyDirectoriesExist(sourcePath, targetPath);
+        
+        try {
+            File sourceDirectory = new File(sourcePath);
+            Matcher fileNameMatcher = fileNamePattern.matcher(StringUtils.EMPTY);
+            int numFilesMoved = 0;
+            
+            LOG.info("Moving files from " + sourcePath + " to " + targetPath);
+            
+            for (File sourceFile : FileUtils.listFiles(sourceDirectory, null, false)) {
+                fileNameMatcher.reset(sourceFile.getName());
+                if (fileNameMatcher.matches()) {
+                    if (LOG.isDebugEnabled()) {
+                        LOG.debug("Moving file: " + sourceFile.getName());
+                    }
+                    moveFileAndMarkAsReady(sourceFile);
+                    numFilesMoved++;
+                }
+            }
+            
+            LOG.info("File move completed.");
+            LOG.info(Integer.toString(numFilesMoved) + " file(s) have been moved to the target directory.");
+        } catch (Exception e) {
+            LOG.error("Encountered error while moving files", e);
+            return false;
+        }
+        
+        return true;
+    }
+
+    protected void verifyDirectoriesExist(String... directoryPaths) {
+        for (String directoryPath : directoryPaths) {
+            File directory = new File(directoryPath);
+            if (!directory.exists() || !directory.isDirectory()) {
+                throw new IllegalStateException("Directory does not exist or does not represent a directory path: " + directoryPath);
+            }
+        }
+    }
+
+    protected void moveFileAndMarkAsReady(File sourceFile) throws IOException {
+        String targetFilePath = buildFilePath(targetPath, sourceFile.getName());
+        File targetFile = new File(targetFilePath);
+        FileUtils.moveFile(sourceFile, targetFile);
+        
+        String doneFilePath = StringUtils.substringBeforeLast(targetFilePath, CUKFSConstants.FILE_EXTENSION_DELIMITER)

```

Comment:
I originally went with FileSystemFileStorageServiceImpl for this, but later decided not to because of the prefixing issues mentioned in my earlier comment. I'll need to investigate further to determine if there's a way to get around the prefix problem.